### PR TITLE
UFP32

### LINKED ISSUE #86: UFP32
### Motivation

Implement an 32-bit unsigned fixed point type.

### PR DESCRIPTION
**Pull request by rostyslavtyshko**
## Type of change

- Bug fix
- New feature
- Improvement (refactoring, restructuring repository, cleaning tech debt, ...)
- Documentation
- Other (describe below)

## Changes

The following changes have been made:

- Change 1
- Change 2

## Notes

- Note 1

## Related Issues

Closes #86

### CODE REVIEW

File: libs/fixed_point/src/ufp32.sw [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+}
+
+impl core::ops::Divide for UFP32 {
+    /// Divide a UFP32 by a UFP32. Panics if divisor is zero.
+    fn divide(self, divisor: Self) -> Self {
+        let zero = UFP32::zero();
+        assert(divisor != zero);
+
+        let denominator: u64 = Self::denominator();
+        // Conversion to U128 done to ensure no overflow happen
```
Comment (bitzoic) [inline libs/fixed_point/src/ufp32.sw:113]:
```suggestion
        // Conversion to U64 done to ensure no overflow happen
```

File: libs/fixed_point/src/ufp32.sw [hunk 2]
```diff
@@ -0,0 +1,263 @@
+library ufp32;
+// A wrapper library around the u32 type for mathematical functions operating with signed 64-bit fixed point numbers.
```
Comment (bitzoic) [inline libs/fixed_point/src/ufp32.sw:2]:
```suggestion
// A wrapper library around the u32 type for mathematical functions operating with unsigned 32-bit fixed point numbers.
```

File: libs/fixed_point/src/ufp32.sw [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+use std::{math::*, u128::U128};
```
Comment (bitzoic) [inline libs/fixed_point/src/ufp32.sw:3]:
```suggestion
use std::math::*;
```
Unused import

**Review by bitzoic — CHANGES_REQUESTED** [review]
Could you update the README to state it now has 3 types [here](https://github.com/FuelLabs/sway-libs/blob/f33fc23bdfd4b57e6be31c43de7181899bf0d28c/libs/fixed_point/README.md?plain=1#L3)? 

There are also 15 build warnings, are they something that you are able to address?

Besides that LGTM 💪

File: libs/fixed_point/src/ufp32.sw (hunk 2 above)
Comment (rostyslavtyshko) [reply libs/fixed_point/src/ufp32.sw]: Fixed

File: libs/fixed_point/src/ufp32.sw (hunk 3 above)
Comment (rostyslavtyshko) [reply libs/fixed_point/src/ufp32.sw]: Fixed

File: libs/fixed_point/src/ufp32.sw (hunk 1 above)
Comment (rostyslavtyshko) [reply libs/fixed_point/src/ufp32.sw]: Fixed

**Review by bitzoic — APPROVED** [review]
:shipit:

**Pull request merged**